README.md

```diff
@@ -498,17 +498,341 @@ group members to eventually converge on a single most preferred epoch.  We have
+   - repeat (incrementing the sequence)
+   
+   
+```mermaid
+flowchart RL
+
+subgraph "feed"
+  direction RL
+d-->c-->b-->a
+end
+
+classDef default stroke:#ccc,fill:#eee,color:#333;
+classDef cluster stroke:#333,fill:none;
+```
+
+The tangle data these messages carry looks like:
+```javascript
+a => { sequence: 1, previous: null }
+b => { sequence: 2, previous: a }
+c => { sequence: 3, previous: b }
+d => { sequence: 3, previous: d }
+```
+(author, signature, and content have been ommited here to make the backlinking 
+pattern clearer)
+
+
+### 6.2. Multi author tangle
+
+In the case of multiple authors, you need to be able to support a DAG which 
+branches and merges (because different authors may contribute concurrently,
+or while offline).
+
+```mermaid
+flowchart RL
+
+M-->X-->B-->A
+M-->Y--->B
+
+classDef default stroke:none,color:#333;
+classDef alice fill:#ccf;
+classDef bob fill:#cfc;
+
+class A,X,M alice
+class B,Y bob
+```
+_Diagram where messages X, Y were both published concurrently (so were unaware
+of one another). M is aware of both X and Y, and extends the tangle from them.
+M is now the new "tip" of the tangle._
+
+The tangle data these messages carry looks like:
+```javascript
+A => { root: null, previous: null }
+B => { root: A,    previous: [A] }
+X => { root: A,    previous: [B] }
+Y => { root: A,    previous: [B] }
+M => { root: A,    previous: [X, Y] }
+```
+
+Where:
+- `root` is the id of the root message of the the tangle
+- `previous` is an Array of message ids of the tip(s) / leading edges of the
+  tangle at the time this message was published.
+    
+Note: the root message cannot include its own id (not known until published), 
+so it sets it's `root` value as `null`, which means "I am a root"
+
+
+<!-- Specificic to private groups -->
+### 6.2.1. Members tangle
+
+The purpose of this tangle is to track the group membership (for a  particular 
+epoch of the group).
+
+We define the members tangle for some id `A` as:
+1. **candate messages**:
+   - the root message with id `A` so long as it has
+     ```javascript
+     content.tangles.members = { root: null, previous: null }
+     ```
+   - any messages where
+     ```javascript
+     content.tangles.members = { root: A, previous: [...] }
+     ```
+2. **recipe**:
+    - a) the graph starts with the root message
+    - b) for the current tip(s) of the tangle
+        - find candidate messages which list that tip in the `previous` field   
+          of their tangle data
+        - check that the graph contains each message in `previous`
+            - if it does, add it to the graph
+            - otherwise set it aside till the graph does contain them
+        - newly added messages to the tangle graph constitue new tips
+    - c) repeat (b) till you cannot grow the tangle further
+
+
+In the context of private groups, the addition to a particular epoch is always
+published to an earlier epoch (or in the case of the first epoch, to the
+Additons feed).
```

Comment:
Only the re-additions. Newly added members will still get their invites on the additions feed. Oh yeah that's another case we have to test. Is that even in the spec? I.e. getting access to old epochs if getting added to a later one.

Made an issue https://github.com/ssbc/ssb-tribes2/issues/76

Reply:
Disagree. Newly added people ... if there are messages `A -> B -> C` should receive invites to :
- `A` - invite published in the Additions feed
- `B` - invite published in `A`
- `C` - invite published in `B`

we do this so that we can easily query the membership tangle in each epoch and not have some funny "special case" for people added to epochs later

Reply:
This is important, and I think @Powersource is right. If I invite my friend to epoch N, I should NOT publish the invite on epoch N-1, because my "group feed" for epoch N-1 is unknown/invisible to anyone who is outside of the group. My friend won't even replicate that group feed, so how could they even see my invite?

Thus it has to be on the additions feed, because this one is publicly known.

Reply:
@mixmix are you saying, when adding a new person, we should give them one invite per epoch? that feels a bit unnecessary when we could just invite them to the latest and point them backwards. i'd say when we're on a new epoch, the old member tangle on the old epoch is no longer representative for the current state of the group. if it was representative, then we'd have to keep updating old epochs forever, which would also be annoying for replication

Reply:
would like this discussion to be resolved before merging, or at least marked as a :fire: TODO for later